Firebase/Core/Private/FIRTimestamp+Internal.h

```diff
@@ -0,0 +1,53 @@
+/*
```

Comment:
I have a couple of questions regarding this file:

* In general, what is the logic behind what should be kept in an internal header? I basically split all but the most basic functionality.
* In the build process, how does the separation into public and private header work? I found out I could include <FirebaseCore/some_header_from_private_folder.h> anywhere in Firestore without any issue. Is there some special build step to make these headers available inside the SDK but unavailable for users, or are private headers always available, and it's just a convention that they shouldn't be used?

Reply:
> I have a couple of questions regarding this file:
> 
>   * In general, what is the logic behind what should be kept in an internal header? I basically split all but the most basic functionality.

The general rule is to reserve the public header for committed public APIs. Everything else goes in the private header.

Some classes within Firestore hide their initializers this way so that they must be constructed using other mechanisms. It's not universally required that initializers be hidden and indeed for plain old data types this makes the types unfriendly.

>   * In the build process, how does the separation into public and private header work? I found out I could include <FirebaseCore/some_header_from_private_folder.h> anywhere in Firestore without any issue. Is there some special build step to make these headers available inside the SDK but unavailable for users, or are private headers always available, and it's just a convention that they shouldn't be used?

The private headers are available at build time within the SDK but aren't published as a part of the built framework. Source-only builds of the SDK have access to these, but source only builds can do whatever they want.